src/core/trulens/core/experimental/__init__.py

```diff
@@ -0,0 +1,441 @@
+from collections import defaultdict
+from enum import Enum
+import functools
+from types import FrameType
+from typing import (
+    Callable,
+    Dict,
+    Generic,
+    Iterable,
+    Mapping,
+    Optional,
+    Set,
+    TypeVar,
+    Union,
+)
+
+import pydantic
+from trulens.core.utils import python as python_utils
+from trulens.core.utils import text as text_utils
+
+T = TypeVar("T")
+
+
+class Feature(str, Enum):
+    """Experimental feature flags.
+
+    Use [TruSession.experimental_enable_feature][trulens.core.session.TruSession.experimental_enable_feature] to enable
+    these features.
+    """
+
+    OTEL_TRACING = "otel_tracing"
+    """OTEL-like tracing.
+
+    !!! Warning
+        This changes how wrapped functions are processed. This setting cannot be
+        changed after any wrapper is produced.
+    """
+
+    @classmethod
+    def _repr_all(cls) -> str:
+        """Return a string representation of all the feature flags."""
+
+        ret = ""
+        for flag in cls:
+            ret += f'  {flag.name} = "{flag.value}"\n'
+
+        return ret
+
+    @classmethod
+    def _missing_(cls, value: str):
+        raise ValueError(
+            f"Invalid feature flag `{value}`. Available flags are:\n{cls._repr_all()}"
+        )
+
+
+class Setting(Generic[T]):
+    """A setting that attains some value and can be locked from changing."""
+
+    def __init__(self, default: T):
+        self.value: T = default
+        """The stored value."""
+
+        self.locked_by: Set[FrameType] = set()
+        """Set of frames (not in trulens) that have locked this value.
+
+        If empty, it has not been locked and can be changed.
+        """
+
+    @property
+    def is_locked(self) -> bool:
```

Comment:
should this be an _internal method instead? Or if user-facing, when would someone us this?

Reply:
Made the whole class private with _.